File: trunk/openrest4js/openrest4js-api/DateTimeWindow.js
function DateTimeWindowFromObj(data)
{
    return new DateTimeWindow(data.start ? OpenrestDateFromObj(data.start) : undefined,
            data.end ? OpenrestDateFromObj(data.end) : undefined);
}

function DateTimeWindow(start, end, available)
{
    this.start = start;
    this.end = end;
    this.available = available;
}
    
DateTimeWindow.prototype.toOpenRestObj = function()
{
    var ret = {};

    if (typeof(this.start) != "undefined") ret['start'] = this.start.toOpenRestObj();
    if (typeof(this.end) != "undefined") ret['end'] = this.end.toOpenRestObj();
    if (typeof(this.available) != "undefined") ret['available'] = this.available;

    return ret;
}

File: openrest4cocoa/openrest4cocoa/openrest4cocoa-api/DateTimeWindow.m
//
//  DateTimeWindow.m
//  openrest4cocoa
//
//  Created by Yoav Amit on 5/4/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "DateTimeWindow.h"


@implementation DateTimeWindow

@synthesize start;
@synthesize end;
@synthesize available;

-(id)initWithDictionary:(NSDictionary*)data
{
    if ((self = [super init]))
    {
        [self setStart:[[[OpenRestDate alloc] initWithDictionary:[data valueForKey:@"start"]] autorelease]];
        [self setEnd:[[[OpenRestDate alloc] initWithDictionary:[data valueForKey:@"end"]] autorelease]];
        [self setAvailable:[[data valueForKey:@"available"] boolValue]];
    }
    return self;
}

-(NSDictionary*)proxyForJson
{    
    NSMutableDictionary* ret = [NSMutableDictionary dictionaryWithCapacity:0];
    [ret setValue:start forKey:@"start"];
    [ret setValue:end forKey:@"end"];
    [ret setValue:[NSNumber numberWithBool:available] forKey:@"available"];
    
    return ret;
}

-(BOOL)isEqual:(id)object
{
    if (object == self) return TRUE;
    
    if (object == NULL) return FALSE;
    if (![object isKindOfClass:[self class]]) return FALSE;
    
    DateTimeWindow* other = (DateTimeWindow*)object;
    
    if ((start != other.start) && (![start isEqual:other.start])) return FALSE;
    if ((end != other.end) && (![end isEqual:other.end])) return FALSE;
    if (available != other.available) return FALSE;
    
    return TRUE;
}

-(NSUInteger)hash
{
    return [start hash] + [end hash] + (available ? 0 : 1);
}

-(void)dealloc
{
    [super dealloc];
}

@end
